### PR TITLE
[ASSIST-741]: Create folders to organize chats with the personal assistant.

### PR DESCRIPTION
**Pull request by mxm1070**
### Ticket(s) or GitHub Issue

https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/741/

### Technical Description

Allow assistant chats to be put into folders.

### Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] New feature (non-breaking change which adds functionality)

### Any deployment steps required?

A deployment step could be a command that needs to be executed or an ENV key that needs to be added, for example.

- [x] No

_______________________________________________

## Before contributing and submitting this PR, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/canyongbs/assistbycanyongbs/blob/main/README.md#contributing).
* [x] Title the PR with the ticket/issue number and a short description of the changes made. Or if no ticket/issue exists, title the PR with a short description of the changes made
* [x] Linked a relevant ticket or issue or describe the issue/feature which this PR resolves/implements.
* [x] Resolved all conflicts, if any.
* [x] Rebased your branch PR on top of the latest upstream `develop` branch.

### CODE REVIEW

File: app-modules/assistant/src/Filament/Pages/PersonalAssistant.php [hunk 1]
```diff
@@ -91,6 +86,30 @@ public function mount(): void
         );
     }
 
+    #[Computed]
```
Comment (Orrison) [inline app-modules/assistant/src/Filament/Pages/PersonalAssistant.php:89]: What is the purpose of refactoring to Livewire computed properties? Does this provide us with some benefit that the original property implementation did not?

Unfortunately, this seems to be one of the biggest causes of the linting errors.

**Review by Orrison — CHANGES_REQUESTED** [review]
Looks like linting is currently failing.

File: composer.json [hunk 2]
```diff
@@ -11,7 +11,6 @@
     "license": "MIT",
     "require": {
         "php": "^8.2",
-        "ext-gd": "*",
```
[the remark this answers is not in the document]
Comment (mxm1070) [reply composer.json:14]: Merge gone wrong.

File: composer.json [hunk 3]
```diff
@@ -64,7 +63,6 @@
         "league/flysystem-aws-s3-v3": "^3.0",
         "league/oauth2-client": "^2.7",
         "livewire/livewire": "^3.0",
-        "maatwebsite/excel": "^3.1",
```
[the remark this answers is not in the document]
Comment (mxm1070) [reply composer.json:67]: Merge gone wrong.

File: composer.json [hunk 4]
```diff
@@ -76,7 +74,6 @@
         "sqids/sqids": "^0.3.1",
         "staudenmeir/eloquent-has-many-deep": "^1.18",
         "stechstudio/filament-impersonate": "^3.5",
-        "tapp/filament-timezone-field": "^3.0",
```
[the remark this answers is not in the document]
Comment (mxm1070) [reply composer.json:79]: Merge gone wrong.

File: app-modules/assistant/src/Filament/Pages/PersonalAssistant.php [hunk 5]
```diff
@@ -219,6 +238,119 @@ public function newChat(): void
...
+
+    public function deleteFolderAction(): Action
+    {
+        return Action::make('deleteFolder')
+            ->size(ActionSize::ExtraSmall)
+            ->requiresConfirmation()
+            ->action(function (array $arguments) {
+                $folder = AssistantChatFolder::find($arguments['folder']);
+
+                //TODO: disassociate for now, maybe delete?
```
[the remark this answers is not in the document]
Comment (mxm1070) [reply app-modules/assistant/src/Filament/Pages/PersonalAssistant.php:301]: Per Product, delete with warning.

File: app-modules/assistant/src/Filament/Pages/PersonalAssistant.php (hunk 1 above)
Comment (mxm1070) [reply app-modules/assistant/src/Filament/Pages/PersonalAssistant.php:93]: I was having problems with losing the sorting anytime the collection changed: https://livewire.laravel.com/docs/properties#eloquent-constraints-arent-preserved-between-requests

I was able to fix the linting with a class property:
```php
/**
 * @property EloquentCollection $chats
 */
```

**Review by Orrison — APPROVED** [review]

**Pull request merged**